### ISSUE TITLE
bug: Postgres table read - `ParseError: Failed to parse 'public.geometry(Geometry,4326)' into <class 'sqlglot.expressions.DataType'>`

### ISSUE THREAD

**Issue by Riezebos** [ibis-project/ibis#8807]
### What happened?

I've been learning ibis and I have to say in general it's amazing, I don't understand why nobody told my about it earlier!

I am trying to select a table from a postgres database, it has a geometry column which causes the error shown in the title and in the output below. I'm not sure if this is a bug or a feature that is not supported.

The error happens when I run:
```
t = con.table("v_teams")
```

There is a column in the table I am querying which is called `lonlat` and has a `geometry` datatype, it has values of POINT and NULL.

Where con is `ibis.backends.postgres.Backend`.

When I run a query that doesn't include the `geometry` column it does work, e.g.:
```
t = con.sql("select id, name from v_teams")
```

### What version of ibis are you using?

8.0.0

### What backend(s) are you using, if any?

postgres

### Relevant log output

```sh
---------------------------------------------------------------------------
ParseError                                Traceback (most recent call last)
File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1077, in Parser.parse_into(self, expression_types, raw_tokens, sql)
   1076 try:
-> 1077     return self._parse(parser, raw_tokens, sql)
   1078 except ParseError as e:

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1116, in Parser._parse(self, parse_method, raw_tokens, sql)
   1115 if self._index < len(self._tokens):
-> 1116     self.raise_error("Invalid expression / Unexpected token")
   1118 self.check_errors()

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1157, in Parser.raise_error(self, message, token)
   1156 if self.error_level == ErrorLevel.IMMEDIATE:
-> 1157     raise error
   1159 self.errors.append(error)

ParseError: Invalid expression / Unexpected token. Line 1, Col: 6.
  public.geometry(Geometry,4326)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Hey @Riezebos -- have you installed the `geospatial` extra?  That might fix this.  So if you installed using `pip`, you would do `pip install -U 'ibis-framework[postgres,geospatial]'`

**Comment by Riezebos** [comment]
Hi @gforsyth, thanks for tip, unfortunately it did not solve the problem, I am getting the exact same error. If I load the data with geopandas.read_postgis(), I don't get any errors.

**Comment by cpcloud** [comment]
@Riezebos Can you show the `CREATE TABLE` statement for the `v_teams` table?

**Comment by Riezebos** [comment]
If I use the following SQL to create a sample table:

```sql
CREATE TABLE v_test (
	id int8 NULL,
	lonlat geometry NULL
);

INSERT INTO volt_team.v_test
(id, lonlat)
select 1 as id, 'POINT (5.7 53.2)' as lonlat;
```

And then use the following code in Python:

```python
import ibis

pg_con = ibis.connect(
    "postgresql://my_server:5432/my_db",
    schema="my_schema",
)

pg_con.sql("select id, lonlat as geom from v_test").execute()
```

This is the full output I get:
```
{
	"name": "ParseError",
	"message": "Failed to parse 'public.geometry' into <class 'sqlglot.expressions.DataType'>",
	"stack": "---------------------------------------------------------------------------
ParseError                                Traceback (most recent call last)
File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1077, in Parser.parse_into(self, expression_types, raw_tokens, sql)
   1076 try:
-> 1077     return self._parse(parser, raw_tokens, sql)
   1078 except ParseError as e:

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1116, in Parser._parse(self, parse_method, raw_tokens, sql)
   1115 if self._index < len(self._tokens):
-> 1116     self.raise_error(\"Invalid expression / Unexpected token\")
   1118 self.check_errors()

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1157, in Parser.raise_error(self, message, token)
   1156 if self.error_level == ErrorLevel.IMMEDIATE:
-> 1157     raise error
   1159 self.errors.append(error)

ParseError: Invalid expression / Unexpected token. Line 1, Col: 6.
  public.geometry

The above exception was the direct cause of the following exception:

ParseError                                Traceback (most recent call last)
Cell In[9], line 1
----> 1 pg_con.sql(\"select id, lonlat as geom from v_test\").execute()

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/base/sql/__init__.py:125, in BaseSQLBackend.sql(self, query, schema, dialect)
    123 query = self._transpile_sql(query, dialect=dialect)
    124 if schema is None:
--> 125     schema = self._get_schema_using_query(query)
    126 else:
    127     schema = sch.schema(schema)

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/base/sql/alchemy/__init__.py:914, in BaseAlchemyBackend._get_schema_using_query(self, query)
    912 def _get_schema_using_query(self, query: str) -> sch.Schema:
    913     \"\"\"Return an ibis Schema from a backend-specific SQL string.\"\"\"
--> 914     return sch.Schema.from_tuples(self._metadata(query))

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/expr/schema.py:118, in Schema.from_tuples(cls, values)
     92 @classmethod
     93 def from_tuples(
     94     cls,
     95     values: Iterable[tuple[str, str | dt.DataType]],
     96 ) -> Schema:
     97     \"\"\"Construct a `Schema` from an iterable of pairs.
     98 
     99     Parameters
   (...)
    116     }
    117     \"\"\"
--> 118     pairs = list(values)
    119     if len(pairs) == 0:
    120         return cls({})

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/postgres/__init__.py:278, in Backend._metadata(self, query)
    276 con.exec_driver_sql(f\"CREATE TEMPORARY VIEW {name} AS {query}\")
    277 try:
--> 278     yield from (
    279         (col, PostgresType.from_string(typestr))
    280         for col, typestr in con.execute(text)
    281     )
    282 finally:
    283     con.exec_driver_sql(f\"DROP VIEW IF EXISTS {name}\")

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/postgres/__init__.py:279, in <genexpr>(.0)
    276 con.exec_driver_sql(f\"CREATE TEMPORARY VIEW {name} AS {query}\")
    277 try:
    278     yield from (
--> 279         (col, PostgresType.from_string(typestr))
    280         for col, typestr in con.execute(text)
    281     )
    282 finally:
    283     con.exec_driver_sql(f\"DROP VIEW IF EXISTS {name}\")

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/postgres/datatypes.py:88, in PostgresType.from_string(cls, type_string)
     86 @classmethod
     87 def from_string(cls, type_string: str) -> PostgresType:
---> 88     return SqlglotPostgresType.from_string(type_string)

File ~/repos/playground/.venv/lib/python3.11/site-packages/ibis/backends/base/sqlglot/datatypes.py:175, in SqlglotType.from_string(cls, text, nullable)
    172 if dtype := cls.unknown_type_strings.get(text.lower()):
    173     return dtype
--> 175 sgtype = sg.parse_one(text, into=sge.DataType, read=cls.dialect)
    176 return cls.to_ibis(sgtype, nullable=nullable)

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/__init__.py:121, in parse_one(sql, read, dialect, into, **opts)
    118 dialect = Dialect.get_or_raise(read or dialect)
    120 if into:
--> 121     result = dialect.parse_into(into, sql, **opts)
    122 else:
    123     result = dialect.parse(sql, **opts)

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/dialects/dialect.py:449, in Dialect.parse_into(self, expression_type, sql, **opts)
    446 def parse_into(
    447     self, expression_type: exp.IntoType, sql: str, **opts
    448 ) -> t.List[t.Optional[exp.Expression]]:
--> 449     return self.parser(**opts).parse_into(expression_type, self.tokenize(sql), sql)

File ~/repos/playground/.venv/lib/python3.11/site-packages/sqlglot/parser.py:1082, in Parser.parse_into(self, expression_types, raw_tokens, sql)
   1079         e.errors[0][\"into_expression\"] = expression_type
   1080         errors.append(e)
-> 1082 raise ParseError(
   1083     f\"Failed to parse '{sql or raw_tokens}' into {expression_types}\",
   1084     errors=merge_errors(errors),
   1085 ) from errors[-1]

ParseError: Failed to parse 'public.geometry' into <class 'sqlglot.expressions.DataType'>"
}
```

In case it helps, this is the database version:
```
PostgreSQL 10.23
PostgreSQL 10.23 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 7.3.1 20180712 (Red Hat 7.3.1-12), 64-bit
```

**Comment by gforsyth** [comment]
Hey @Riezebos -- sorry for the radio silence here, this issue dropped off my plate.

This is fixed on `main` and will be available in the soon-to-be-released 9.0, or you can try out a pre-release by installing with `python -m pip install -U --pre 'ibis-framework[postgres,geospatial]'`

```python
[ins] In [1]: import ibis

[ins] In [2]: con = ibis.postgres.connect(
         ...:     user="postgres",
         ...:     password="postgres",
         ...:     host="0.0.0.0",
         ...:     port=5432,
         ...:     database="ibis_testing",
         ...: )

[ins] In [3]: query = """CREATE TABLE v_test (
         ...: ^Iid int8 NULL,
         ...: ^Ilonlat geometry NULL
         ...: );
         ...: 
         ...: INSERT INTO v_test
         ...: (id, lonlat)
         ...: select 1 as id, 'POINT (5.7 53.2)' as lonlat;"""

[ins] In [4]: with con.raw_sql(query) as c:
         ...:     pass

[ins] In [5]: con.table("v_test")
Out[5]: 
DatabaseTable: v_test
  id     int64
  lonlat geospatial:geometry

[ins] In [6]: ibis.options.interactive = True

[ins] In [7]: con.table("v_test")
Out[7]: 
┏━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┓
┃ id    ┃ lonlat               ┃
┡━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
│ int64 │ geospatial:geometry  │
├───────┼──────────────────────┤
│     1 │ <POINT (5.7 53.2)>   │
└───────┴──────────────────────┘
```

Thanks again for the report and sorry for the delay in responding!
